base/dates/types.jl

```diff
@@ -58,10 +58,11 @@ immutable Date <: TimeType
     Date(instant::UTInstant{Day}) = new(instant)
 end
 
-# Fallback constructors
-_c(x) = convert(Int64,x)
-DateTime(y,m=1,d=1,h=0,mi=0,s=0,ms=0) = DateTime(_c(y),_c(m),_c(d),_c(h),_c(mi),_c(s),_c(ms))
-Date(y,m=1,d=1) = Date(_c(y),_c(m),_c(d))
+# Time is a nanosecond precision representation of hours:minutes:seconds.fractional
+immutable Time <: TimeType
+    instant::Nanosecond
+    Time(instant::Nanosecond) = new(instant)
```

Comment:
Hmm, couldn't someone create a `Time` with a negative nanoseconds or greater than or equal to 24:00:00? Do we need to enforce that here?